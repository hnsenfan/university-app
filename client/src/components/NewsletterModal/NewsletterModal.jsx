import { useCallback } from 'react'
import PropTypes from 'prop-types'

import Modal from "../Modal/Modal"
import { END_POINTS } from '../../api'

const NewsletterModal = ({ show, handleClose }) => {
  const handleSubmitSubscription = useCallback(async() => {
    await END_POINTS.patchNewsletterSubscription(true)
    handleClose()
  })
  return (
    <Modal show={show} handleClose={handleClose}>
      <h1>Join our mailing list!</h1>
      <p>We really want you to want us. Sign up to receive email updates on new features, university, specials, and more</p>
      <button className='btn-primary' onClick={handleSubmitSubscription}>Yes, subscribe me to this list</button>
      <p>You won&apos;t be subscribed if you don&apos;t click the confirmation button above.</p>
    </Modal>
  )
}

NewsletterModal.propTypes = {
  show: PropTypes.bool,
  handleClose: PropTypes.func
}

export default NewsletterModal