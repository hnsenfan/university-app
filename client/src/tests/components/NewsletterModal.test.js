import { shallow } from 'enzyme'
import NewsletterModal from '../../components/NewsletterModal/NewsletterModal'

const wrapper = shallow(<NewsletterModal />)

test('should test NewsletterModal component', () => {
 expect(wrapper).toMatchSnapshot()
})

test('should have a subscribe button', () => {
  expect(wrapper.find('.btn-primary')).toHaveLength(1)
})
