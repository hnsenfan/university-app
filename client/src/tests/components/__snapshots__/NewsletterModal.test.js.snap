// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test NewsletterModal component 1`] = `
<Modal>
  <h1>
    Join our mailing list!
  </h1>
  <p>
    We really want you to want us. Sign up to receive email updates on new features, university, specials, and more
  </p>
  <button
    className="btn-primary"
    onClick={[Function]}
  >
    Yes, subscribe me to this list
  </button>
  <p>
    You won't be subscribed if you don't click the confirmation button above.
  </p>
</Modal>
`;
